spec/controllers/registrations_controller_spec.rb

```diff
@@ -37,12 +37,58 @@
             name: u.name,
             email: u.email,
             password: u.password,
-            password_confirmation: u.password_confirmation
+            password_confirmation: u.password
           }
         }
       end.to change(User, :count).by 1
 
       expect(flash[:notice]).to eq I18n.t('devise.registrations.signed_up_but_unconfirmed')
     end
   end
+
+  context 'email confirmation enabled' do
+    before do
+      stub_local_login_setting(email_confirmation: true)
+    end
+
+    it 'allows users to register' do
+      u = build(:user)
+
+      expect do
+        post :create, params: {
+          user: {
+            name: u.name,
+            email: u.email,
+            password: u.password,
+            password_confirmation: u.password
+          }
+        }
+      end.to change(User, :count).by 1
+
+      expect(flash[:notice]).to eq I18n.t('devise.registrations.signed_up_but_unconfirmed')
+    end
+  end
+
+  context 'email confirmation disabled' do
+    before do
+      stub_local_login_setting(email_confirmation: false)
+    end
+
+    it 'allows users to register without confirming email' do
+      u = build(:user)
```

Comment:
Does `let(:u) { build(:user) }` work instead of declaring it here? (The `let` would have to be moved above the `it` statement.